docs/advanced-features/tabulator.md

```diff
@@ -0,0 +1,134 @@
+<!-- markdownlint-disable-next-line first-line-h1 -->
+> NOTE: This feature requires Quilt stack version 1.55.0 or higher
+
+Tabulator aggregates tabular data objects across multiple packages using AWS
+Athena. Admins define schemas and data sources for CSV, TSV, or Parquet files,
+enabling users to run SQL queries directly on the contents of Quilt packages.
+You can even use named capture groups to extract additional columns from
+the logical key and package name.
+
+The configuration is written in YAML and managed using the
+`quilt3.admin.tabulator`
+[APIs](../api-reference/Admin.md#quilt3.admin.tabulator) or via the
+Quilt Admin UI:
+
+![Admin UI for setting Tabulator configuration](../imgs/admin-tabulator-config.png)
+
+## Configuration
+
+Each Tabulator configuration is written in YAML, following the structure
+outlined below.
+
+### Example
+
+```yaml
+schema:
+  - name: name  # must match ^[a-z_][a-z0-9_]*$
+    type: STRING  # usually BOOLEAN, INT, FLOAT, DOUBLE, STRING, DATE, TIMESTAMP
+  - name: length
+    type: INT
+  - name: effective_length
+    type: FLOAT
+  - name: tpm
+    type: FLOAT
+  - name: num_reads
+    type: FLOAT
+source:
+  type: quilt-packages  # currently the only supported type
+  package_name: "^ccle/(?<date>[^_]+)_(?<study_id>[^_]+)_nfcore_rnaseq$"
+  logical_key: "salmon/(?<sample_id>[^/]+)/quant*\\.genes\\.sf$"
+parser:
+  format: csv  # or `parquet`
+  delimiter: "\t"
+  header: true
+```
+
+A more complex example is available in [this yaml file](./tabulator-example.yaml).
+
+1. **Schema**: The schema defines the columns in the table. Each column must
+   have a name and a type. The name must match the regular expression
```

Comment:
this is only valid for CSVs. for parquet the names and the types must match (tho you can select a subset of columns)

Reply:
Does that mean Parquet already ensures all column names match that regex?

Reply:
well, no.
two things here:
1. select is case-insensitive, so even if parquet has columns with a different case in the name it will work anyways
2. there's currently no way of selecting parquet columns with incompatible names (for that we may consider to e.g. adding a new optional "source" field to the column definition format)